goenvtemplator.go

```diff
@@ -131,32 +153,26 @@ func generateTemplates(
 	return nil
 }
 
-var (
-	v            int
-	buildVersion string = "Build version was not specified."
-)
-
 func main() {
 	var tmpls templatesPaths
-	flag.Var(&tmpls, "template", "Template (/template:/dest). Can be passed multiple times.")
-	var debugTemplates bool
-	flag.BoolVar(&debugTemplates, "debug-templates", false, "Print processed templates to stdout.")
 	var doExec bool
-	flag.BoolVar(&doExec, "exec", false, "Activates exec by command. First non-flag arguments is the command, the rest are it's arguments.")
 	var printVersion bool
-	flag.BoolVar(&printVersion, "version", false, "Prints version.")
 	var envFileList envFiles
-	flag.Var(&envFileList, "env-file", "Additional file with environment variables. Can be passed multiple times.")
 	var delimLeft string
-	flag.StringVar(&delimLeft, "delim-left", "", "Override default left delimiter {{.")
 	var delimRight string
+	var engine string
+
+	flag.Var(&tmpls, "template", "Template (/template:/dest). Can be passed multiple times.")
+	flag.BoolVar(&DEBUG, "debug-templates", false, "Print processed templates to stdout.")
+	flag.BoolVar(&doExec, "exec", false, "Activates exec by command. First non-flag arguments is the command, the rest are it's arguments.")
+	flag.BoolVar(&printVersion, "version", false, "Prints version.")
+	flag.Var(&envFileList, "env-file", "Additional file with environment variables. Can be passed multiple times.")
+	flag.StringVar(&delimLeft, "delim-left", "", "Override default left delimiter {{.")
```

Comment:
You should at least mention, that `delims` are used only with default engine. And maybe we should think a way how to make this more general.

Reply:
Maybe a custom help for every engine? Every engine could have its own options.